dpgen2/entrypoint/args.py

```diff
@@ -78,6 +78,7 @@ def dp_train_args():
             "init_models_paths",
             list,
```

Comment:
```suggestion
            [list, None],
```

Reply:
```python
from dargs import Argument

def make_arguments():
    arg_foo = Argument("foo", int, optional=True, default=None)
    return [arg_foo]

base = Argument("base", dict, make_arguments())
data = base.normalize_value({'foo' : None})
print(data)
data = base.normalize_value({})
print(data)
```
the result is 
```
{'foo': None}
{'foo': None}
```

Reply:
https://github.com/deepmodeling/dargs/blob/11d3fc9ec197924534d9612a155dbfa9d3368583/dargs/dargs.py#L214

This is because the type of default will be added to dtype automatically. However, it's better to explicitly set it.

Reply:
In this project we usually write 
```
Optional[int] = None
```
as type hint and default value, not 
```
Union[int, None] = None
```
The dargs definition is more consistent with the first way.